packages/zoe/src/objArrayConversion.js

```diff
@@ -20,8 +22,9 @@ export const arrayToObj = (array, keys) => {
     array.length === keys.length,
     details`array and keys must be of equal length`,
   );
-  /** @type {{[Keyword: U]: T}} */
-  const obj = {};
+  const obj =
+    /** @type {Record<U, T>} */
```

Comment:
Why can't this go above the `const obj =` ?

Reply:
If I do it after the assignment (i.e. above the `const`), I get `Type '{}' is not assignable to type 'Record<U, T>'`.  The default narrowing of `{}` is inferred as a type that can't be cast to the Record.  If I do it as a cast around the empty object (`{}`), then it can still be coerced.